src/main/java/org/b612foundation/adam/astro/AstroUtils.java

```diff
@@ -7,17 +7,26 @@
 import java.time.temporal.ChronoUnit;
 import java.time.temporal.JulianFields;
 
-import static org.b612foundation.adam.astro.AstroConstants.DAY_TO_SEC;
+import static org.b612foundation.adam.astro.AstroConstants.*;
 
 public class AstroUtils {
 
     public static LocalDateTime localDateTimefromMJD(double mjd) {
         long dayComponent = (long)mjd;
-        double timeComponentSeconds = (mjd - dayComponent) * DAY_TO_SEC;
-        long timeSeconds = (long)timeComponentSeconds;
-        long timeNanoSeconds = (long)(timeComponentSeconds - timeSeconds);
-        LocalDate date = LocalDate.MIN.with(JulianFields.MODIFIED_JULIAN_DAY, dayComponent);
-        LocalTime time = LocalTime.MIDNIGHT.plusSeconds(timeSeconds).plusNanos(timeNanoSeconds);
+        double timeComponentDays = (mjd - dayComponent);
+        double hoursFull = timeComponentDays * DAY_TO_HR;
+        long hoursComponent = (long) hoursFull;
```

Comment:
Here and everywhere, might be a good idea to explicitly say `Math.floor(hoursFull)`.

Reply:
Yeah doesn't hurt to be explicit.  It's still a double after floor so need to still cast to long.